python/GafferUI/AnimationEditor.py

```diff
@@ -868,4 +879,186 @@ def __checkBoxStateForTieMode( self, mode ) :
 		commonMode = self.getTieModeForSelectedKeys()
 		return None if commonMode is None else commonMode == mode
 
+# Private implementation - curve widget
+class _CurveWidget( GafferUI.GridContainer ) :
+
+	from collections import namedtuple
+	Connections = namedtuple( "Connections", ("color") )
+
+	def __init__( self ) :
+
+		GafferUI.GridContainer.__init__( self, spacing=4, borderWidth=4 )
+
+		# tool tips
+		colorToolTip = "# Color\n\nThe color of the currently selected curves.\n"
+
+		# create labels
+		colorLabel = GafferUI.Label( text="Color", toolTip=colorToolTip )
+
+		# create editors
+		self.__colorSwatch = GafferUI.ColorSwatch( useDisplayTransform=False, toolTip=colorToolTip )
+		self.__colorSwatch._qtWidget().setFixedHeight( 18 )
+		self.__colorInvalid = GafferUI.TextWidget( text="---", editable=False )
+		self.__colorInvalid.setEnabled( False )
+
+		# setup editor connections
+		self.__colorSwatch.buttonPressSignal().connect( Gaffer.WeakMethod( self.__swatchButtonPress ), scoped = False )
+		self.__colorSwatch.buttonReleaseSignal().connect( Gaffer.WeakMethod( self.__swatchButtonRelease ), scoped = False )
+		self.__colorSwatch.dragBeginSignal().connect( Gaffer.WeakMethod( self.__swatchDragBegin ), scoped = False )
+		self.__colorSwatch.dragEndSignal().connect( Gaffer.WeakMethod( self.__swatchDragEnd ), scoped = False )
+		self.__colorSwatch.contextMenuSignal().connect( Gaffer.WeakMethod( self.__swatchContextMenu ), scoped = False )
+
+		# layout widgets
+		alignment = ( GafferUI.HorizontalAlignment.Right, GafferUI.VerticalAlignment.Center )
+		self.addChild( colorLabel, index=( 0, 0 ), alignment=alignment )
+		self[ 1:3, 0 ] = self.__colorSwatch
+
+		# curve connections
+		self.__connections = {}
+
+		# undo queue state
+		self.__lastChangedReasonColor = None
+		self.__mergeGroupIdColor = 0
+
+		# initial color of curves
+		self.__initialColors = {}
+
+	def connect( self, curve ) :
+		if curve not in self.__connections :
+			self.__connections[ curve ] = _CurveWidget.Connections(
+				color = curve.colorChangedSignal().connect( Gaffer.WeakMethod( self.__colorChanged ), scoped = False ) )
+
+	def disconnect( self, curve ) :
+		if curve in self.__connections :
+			for connection in self.__connections[ curve ] :
+				connection.disconnect()
+			del self.__connections[ curve ]
+
+	def update( self ) :
+		self.__updateColor()
+
+	def __swatchContextMenu( self, *unused ) :
+
+		menuDefinition = IECore.MenuDefinition()
+		menuDefinition.append(
+			"/Set Default", {
```

Comment:
Just "Default" here please, to match the equivalent menu item in the NodeEditor. "Set Default" implies that we're changing the default.

Reply:
done